apis/src/responses/challenge.rs

```diff
@@ -70,5 +72,77 @@ impl ChallengeResponse {
             band_lower: challenge.band_lower,
         })
     }
+
+}
+}
+}
+
+fn is_compatible(challenge: &ChallengeResponse, user: &str, details: &ChallengeDetails) -> bool {
+    details.game_type == GameType::from_str(&challenge.game_type).unwrap()
+        && details.rated == challenge.rated
+        && details.time_mode == challenge.time_mode
+        && details.time_base == challenge.time_base
+        && details.time_increment == challenge.time_increment
+        && match details.color_choice {
+            ColorChoice::Random => true,
+            ColorChoice::White => {
+                challenge.color_choice == ColorChoice::Black
+                    || challenge.color_choice == ColorChoice::Random
```

Comment:
Don't think we should match people looking for Random with people looking for a certain color. I might be fine playing Random but wouldn't accept a challenge where it's certain I'll play Black for example